plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMinioOrcConnectorTest.java

```diff
@@ -163,4 +164,16 @@ public void testDropAmbiguousRowFieldCaseSensitivity()
                 .hasMessageContaining("Error opening Iceberg split")
                 .hasStackTraceContaining("Multiple entries with same key");
     }
+
+    @Override
+    protected Optional<TimestampPrecisionTestSetup> filterTimestampPrecisionOnCreateTableAsSelectProvider(TimestampPrecisionTestSetup setup)
+    {
+        if (setup.sourceValueLiteral().equals("TIMESTAMP '1969-12-31 23:59:59.999999499999'")) {
+            return Optional.of(setup.withNewValueLiteral("TIMESTAMP '1970-01-01 00:00:00.999999'"));
+        }
+        if (setup.sourceValueLiteral().equals("TIMESTAMP '1969-12-31 23:59:59.9999994'")) {
+            return Optional.of(setup.withNewValueLiteral("TIMESTAMP '1970-01-01 00:00:00.999999'"));
+        }
+        return Optional.of(setup);
```

Comment:
@findepi : Do you know if this is something ORC related?

Reply:
not sure i have context. this being what exactly?

Reply:
Providing a timestamp like `TIMESTAMP '1969-12-31 23:59:59.9999994` and `TIMESTAMP '1969-12-31 23:59:59.999999499999'` as an input value for a CTAS statement results in a Iceberg table with value `TIMESTAMP '1970-01-01 00:00:00.999999`.

Is this expected for ORC files?

Reply:
this looks like a bug. i remember there were some +/-1s for at-the-epoch values in ORC, but @dain  will know more

are you able to reproduce this without coercions, i.e. with direct insert of a timestamp(6) value into Iceberg?

Reply:
This seems also prevalent in Hive:

```
trino:default> CREATE table r AS SELECT TIMESTAMP '1969-12-31 23:59:59.999' a;
CREATE TABLE: 1 row

Query 20230920_082205_00009_62iqv, FINISHED, 1 node
Splits: 18 total, 18 done (100,00%)
0,42 [0 rows, 0B] [0 rows/s, 0B/s]

trino:default> select * from r;
            a
-------------------------
 1970-01-01 00:00:00.999
(1 row)

Query 20230920_082212_00010_62iqv, FINISHED, 1 node
Splits: 1 total, 1 done (100,00%)
0,31 [1 rows, 259B] [3 rows/s, 849B/s]
```